library/azure_rm_resource.py

```diff
@@ -260,6 +258,19 @@ def exec_module(self, **kwargs):
 
             if orphan is not None:
                 self.url += '/' + orphan
+
+        # if api_version was not specified, get latest one
+        if not self.api_version:
+            # extract provider and resource type
+            provider = self.url.split("/providers/")[1].split("/")[0]
+            resourceType = self.url.split(provider + "/")[1].split("/")[0]
```

Comment:
what if it's url has subresource like:
/subscriptions/xxx/resourceGroups/api2018/providers/Microsoft.Network/networkInterfaces/acctni
does  it  work?

Reply:
yes

Reply:
if there's no "providers/xx/xx" then we can't get api version, so it should fail